player/lru_cache.go

```diff
@@ -0,0 +1,149 @@
+package player
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/lbryio/lbry.go/v2/stream"
+)
+
+// LRUCacheOpts contains options for a cache. Size is max size in bytes.
+type LRUCacheOpts struct {
+	Path          string
+	Size          uint64
+	SweepInterval time.Duration
+}
+
+type lruCache struct {
+	storage *fsStorage
+	lru     *lru.Cache
+}
+
+// InitLRUCache initializes a LRU cache for chunks.
+func InitLRUCache(opts *LRUCacheOpts) (ChunkCache, error) {
+	storage, err := initFSStorage(opts.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	if opts.Size == 0 {
+		opts.Size = defaultMaxCacheSize
+	}
+
+	onEvicted := func(key interface{}, value interface{}) {
+		storage.remove(value)
+	}
+
+	lru, err := lru.NewWithEvict(int(opts.Size/ChunkSize), onEvicted)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &lruCache{storage, lru}
+
+	Logger.Infof("LRU cache of %vGB initialized at %v", opts.Size/1024/1024/1024, opts.Path)
```

Comment:
I think that the implicit type is int, you might want to use 1024.0 instead

Reply:
That's alright, integer precision is fine in this case.